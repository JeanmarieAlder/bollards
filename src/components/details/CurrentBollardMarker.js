import React, { useEffect, useRef } from 'react';
import { Marker, Popup } from 'react-leaflet';
import 'leaflet';
import { getMarkerByType } from '../../utils/map/MarkerUtils';

const CurrentBollardMarker = ({ bollard }) => {
    const leafletRef = useRef();
    useEffect(() => {
        leafletRef.current.openPopup();
    }, [bollard])
    return (
        <Marker ref={leafletRef} position={[bollard.b_lat, bollard.b_lng]} icon={getMarkerByType(bollard.b_type)}>
            <Popup>
                <h2>No {bollard.b_number}{bollard.b_letter}</h2>
            </Popup>
        </Marker>
    )
};

export default CurrentBollardMarker;