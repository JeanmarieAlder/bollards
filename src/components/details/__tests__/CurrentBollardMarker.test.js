import React from 'react';
import { MapContainer } from 'react-leaflet';
import renderer from 'react-test-renderer';
import CurrentBollardMarker from '../CurrentBollardMarker';

//jest.mock('leaflet');

test("Component displays correctly", () => {
    const component = renderer.create(
        <div style={{height:'100%'}}>
            <MapContainer style={{height:'100%'}}>
                <CurrentBollardMarker bollard={{
                    b_lat: "46.1", 
                    b_lng: "6.9", 
                    b_type: "Forest", 
                    b_number: 1,
                    b_letter: "a"
                }}/>
            </MapContainer>
        </div>
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
});