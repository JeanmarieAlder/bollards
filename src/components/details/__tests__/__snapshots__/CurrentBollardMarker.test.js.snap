// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component displays correctly 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
      }
    }
  />
</div>
`;
